bodhi/server/services/releases.py

```diff
@@ -332,3 +337,75 @@ def save_release(request):
     request.db.flush()
 
     return r
+
+
+@critpath_packages.get(permission='edit', renderer='json',
+                       error_handler=bodhi.server.services.errors.json_handler,
+                       validators=(colander_body_validator, validate_release_name)
+                       )
+def get_critpath_packages(request):
+    """
+    Get a list of critical path packages for a specific release.
+
+    Args:
+        request (pyramid.request): The current request.
+    Returns:
+        dict: The dictionary {'packages': critpath_packages}.
+    """
+    data = request.validated
+
+    release = data.pop('release', None)
+
+    return dict(packages=release.get_critpath_pkgs())
+
+
+@critpath_packages.put(schema=bodhi.server.schemas.UpdateCriticalPackages,
+                       permission='edit', renderer='json',
+                       error_handler=bodhi.server.services.errors.json_handler,
+                       validators=(colander_body_validator,
+                                   validate_release_name,
+                                   validate_packages)
+                       )
+def add_critpath_packages(request):
+    """
+    Add a list of critical path packages to a specific release.
+
+    Args:
+        request (pyramid.request): The current request.
+    Returns:
+        dict: The dictionary {'packages': updated_critpath_packages}.
+    """
+    data = request.validated
+
+    release = data.pop('release', None)
+    packages = data.pop('packages', None)
```

Comment:
Similar here, you can drop all three of the above lines and simplify things a bit.